scripts/zones/Aydeewa_Subterrane/npcs/Excavation_Site.lua

```diff
@@ -0,0 +1,59 @@
+-----------------------------------
+-- Area: Aydeewa Subterrane
+--  NPC: Excavation Site (Olduum Ring quest)
+-- @pos 390 1 349 68
+-----------------------------------
+package.loaded["scripts/zones/Aydeewa_Subterrane/TextIDs"] = nil;
+-----------------------------------
+ 
+require("scripts/globals/keyitems");
+require("scripts/zones/Aydeewa_Subterrane/TextIDs");
+ 
+-----------------------------------
+-- onTrade Action
+-----------------------------------
+ 
+function onTrade(player,npc,trade)
+    if (player:hasKeyItem(DKHAAYAS_RESEARCH_JOURNAL)) then -- If no journal, just stop right here
+        if (trade:hasItemQty(605,1) and trade:getItemCount() == 1) then -- Trade Pickaxe
+            local keyItems =
+            {
+                ELECTROCELL,
+                ELECTROPOT,
+                ELECTROLOCOMOTIVE,
+            }
+            local KI = math.random(1,3);
+            if (player:hasKeyItem(keyItems[KI]) == false) then
```

Comment:
```The name of the key item will start with "Electro" and there are several different ones, but you can only get 1.
(<http://ffxi.allakhazam.com/db/quests.html?fquest=715>)```
check they don't have *any* of the keyitems here (just check them all individually), the adding of the keyitem is fine as it is though :)